ckanext/activity/subscriptions.py

```diff
@@ -0,0 +1,179 @@
+    return {
+        tk.signals.action_succeeded: [
+            {'sender': 'bulk_update_public', 'receiver': bulk_changed},
+            {'sender': 'bulk_update_private', 'receiver': bulk_changed},
+            {'sender': 'bulk_update_delete', 'receiver': bulk_changed},
+
+            {'sender': 'package_create', 'receiver': package_changed},
+            {'sender': 'package_update', 'receiver': package_changed},
+            {'sender': 'package_delete', 'receiver': package_changed},
+
+            {'sender': 'group_create', 'receiver': group_or_org_changed},
+            {'sender': 'group_update', 'receiver': group_or_org_changed},
+            {'sender': 'group_delete', 'receiver': group_or_org_changed},
+
+            {'sender': 'organization_create',
+             'receiver': group_or_org_changed},
+            {'sender': 'organization_update',
+             'receiver': group_or_org_changed},
+            {'sender': 'organization_delete',
+             'receiver': group_or_org_changed},
+
+            {'sender': 'user_create', 'receiver': user_changed},
+            {'sender': 'user_update', 'receiver': user_changed},
+        ]
+    }
+
+
+# action, context, data_dict, result
+def bulk_changed(sender: str, **kwargs: Any):
+    for key in ("context", "data_dict"):
+        if key not in kwargs:
+            log.warning("Activity subscription ignored")
+            return
+
+    context: types.Context = kwargs["context"]
+    datasets = kwargs["data_dict"].get("datasets")
+    model = context["model"]
+
+    user = context['user']
+    user_obj = model.User.get(user)
+    if user_obj:
+        user_id = user_obj.id
+    else:
+        user_id = 'not logged in'
+    for dataset in datasets:
+        entity = model.Package.get(dataset)
+        assert entity
+
+        activity = Activity.activity_stream_item(entity, 'changed', user_id)
+        model.Session.add(activity)
+
+    model.Session.commit()
+
+
+# action, context, data_dict, result
+def package_changed(sender: str, **kwargs: Any):
+    for key in ("result", "context", "data_dict"):
+        if key not in kwargs:
+            log.warning("Activity subscription ignored")
+            return
+
+    type_ = "new" if sender == "package_create" else "changed"
+
+    context: types.Context = kwargs["context"]
+    result: types.ActionResult.PackageUpdate = kwargs["result"]
+    data_dict = kwargs["data_dict"]
+
+    if not result:
+        id_ = data_dict["id"]
+    elif isinstance(result, str):
+        id_ = result
+    else:
+        id_ = result["id"]
+
+    pkg = context["model"].Package.get(id_)
+    assert pkg
+
+    if pkg.private:
+        return
+
+    user_obj = context["model"].User.get(context["user"])
+    if user_obj:
+        user_id = user_obj.id
+    else:
+        user_id = 'not logged in'
+
+    activity = Activity.activity_stream_item(pkg, type_, user_id)
+    context["session"].add(activity)
+    context["session"].commit()
```

Comment:
`activity_create` action requires an existing user ID. Such code bypasses this restriction. If you check a few lines above, you'll see `user_id = 'not logged in'` line, which allows changing a package anonymously(via `ignore_auth` I suppose). 

Even though it looks like a really bad idea, we cannot just require an existing user here. I definitely have seen `not logged in` in the activity streams, which means that such a change can potentially break existing workflows.

So it's one of those "bugs and problems" that I noticed and decided to keep in order to avoid behavioral changes as much as possible.